lib/quiet_quality/executors/base_executor.rb

```diff
@@ -22,6 +22,14 @@ def any_failure?
         pipelines.any?(&:failure?)
       end
 
+      def successful_outcomes
+        @_successful_outcomes ||= outcomes.select(&:success?)
+      end
+
+      def failed_outcomes
+        @_failed_outcomes ||= outcomes.select(&:failure?)
+      end
```

Comment:
I think we need tests for these in the `spec/quiet_quality/executors/executor_examples.rb` shared examples (there are no tests for the abstract base-class)